zerver/actions/reminders.py

```diff
@@ -0,0 +1,40 @@
+import datetime
+
+from zerver.actions.message_send import check_message
+from zerver.actions.scheduled_messages import do_schedule_messages
+from zerver.lib.addressee import Addressee
+from zerver.lib.reminders import access_message_for_reminder
+from zerver.models import Client, ScheduledMessage, UserProfile
+
+
+def schedule_reminder_for_message(
+    current_user: UserProfile,
+    client: Client,
+    message_id: int,
+    deliver_at: datetime.datetime,
+) -> int:
+    message = access_message_for_reminder(current_user, message_id)
+    # Even though reminder will be sent from NOTIFICATION_BOT, we still
+    # set current_user as the sender here to help us make the permission checks easier.
+    addressee = Addressee.for_user_profile(current_user)
+    # This assumes that any logged in user can send a message to self.
```

Comment:
It doesn't assume -- it will throw an exception? Maybe

> This can raise an exception in the unlikely event that the current user cannot DM themself.

Reply:
Changed.